### PR TITLE
Bake additional cert into Jenkins controller image

### PR DESCRIPTION
**Pull request by jlebon**
Just like we did for the agent image, play a similar trick to have the
additional cert baked into the controller image. This is required if we
need to fetch resources from HTTPS URLs covered by the root CA in code
that runs directly on the controller. This situation comes up in RHCOS
where we want to be able to load the pipecfg from an internal repo early
on in jobs.

### CODE REVIEW

**Comment by jlebon** [comment]
I like the `with-cert` idea. It's a bit wordy, but clear. I find the `layer` wording confusing though. I think it'd be clearer if we remain consistent with the canonical OpenShift images? The two images are `jenkins` and `jenkins-agent-base`. So WDYT about:
- `jenkins-with-cert`
- `jenkins-agent-base-with-cert`
?

IOW, this is saying "these are the same as the regular OpenShift images, except they have a cert added". And then we rename our S2I buildconfig and output to `jenkins-s2i`. It's confusing to have it named just `jenkins`.

And I'll add a diagram somewhere showing how they're connected.

**Comment by dustymabe** [comment]
yeah let's try that and see what it looks like in the next upload.

**Comment by jlebon** [comment]
Updated this now with better naming and a diagram!

**Comment by dustymabe** [comment]
Looks much better now. Thanks for the diagram!

**Review by dustymabe — APPROVED** [review]
LGTM
